### PR TITLE
fix(#84) Prevent Duplication of Alias

### PR DESCRIPTION
**Pull request by Danko-chan**
Add an additional endpoint for alias status :-  /minify/alias/:alias/status
error handler for duplicate alias

### CODE REVIEW

File: backend/routers/minify.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ router.get('/alias/:alias', controller.getURLData)
 
 router.post('/add', controller.addURL)
 router.post('/add/custom', auth, controller.addURLAuthed)
+router.post('/alias/:alias/status', controller.getAliasStatus)
```
Comment (BRAVO68WEB) [inline backend/routers/minify.js:13]: Use as get request instead of post

File: backend/controllers/minify.js [hunk 2]
```diff
@@ -41,9 +41,34 @@ module.exports.findUrlById = async (req, res) => {
       })
 }
 
+module.exports.getAliasStatus = async (req, res) => {
+   try{
+      const alias = req.body.alias;
```
Comment (BRAVO68WEB) [inline backend/controllers/minify.js:46]: use params  instead of body

File: backend/controllers/minify.js [hunk 3]
```diff
@@ -41,9 +41,34 @@ module.exports.findUrlById = async (req, res) => {
...
+      const aliasStatus = await minifed_urls.findOne({ alias });
+
+      if(!aliasStatus){       
+         return res.status(200).json("success: true");
```
Comment (BRAVO68WEB) [inline backend/controllers/minify.js:50]: Return  like this .json({success: true})

File: backend/controllers/minify.js [hunk 4]
```diff
@@ -41,9 +41,34 @@ module.exports.findUrlById = async (req, res) => {
...
+      }else{
+         return res.status(400).json("success: false")
```
Comment (BRAVO68WEB) [inline backend/controllers/minify.js:52]: Return  like this .json({success: false})

File: backend/controllers/minify.js [hunk 5]
```diff
@@ -41,9 +41,34 @@ module.exports.findUrlById = async (req, res) => {
...
+}
+
 module.exports.addURL = async (req, res) => {
    const alias = nanoid(5)
    const minifiedUrl = base_url + alias
+   
+   try{
+   const aliasPresent = await minifed_urls.findOne({ alias });
+   if(aliasPresent){      
+      return res.status(400).json("success: false");
```
Comment (BRAVO68WEB) [inline backend/controllers/minify.js:66]: Return  like this .json({success: false})

**Review by BRAVO68WEB — CHANGES_REQUESTED** [review]
@Danko-chan Fix these and its all fine

File: backend/controllers/minify.js (hunk 5 above)
Comment (Danko-chan) [reply backend/controllers/minify.js]: done

**Comment by BRAVO68WEB** [comment]
Is the code even working at your end?
Have you tested it?

**Comment by Danko-chan** [comment]
> Is the code even working at your end? Have you tested it?

sorry, now it's working

**Review by BRAVO68WEB — CHANGES_REQUESTED** [review]
Status always returns true weather alias is available or not.

**Comment by Danko-chan** [comment]
> Status always returns true weather alias is available or not.

Its working fine on my end, if alias present it is true and if not false

**Comment by BRAVO68WEB** [comment]
No, It return true for every alias

**Comment by BRAVO68WEB** [comment]
Can you send a screenshot from your side with both true and false state for status

**Comment by Danko-chan** [comment]
you need to sent a get req like this http://localhost:5000/minify/alias/Zh2fzQ/status with no colon 

not like this  http://localhost:5000/minify/alias/:Zh2fzQ/status

otherwise u will always get true

**Comment by BRAVO68WEB** [comment]
> you need to sent a get req like this http://localhost:5000/minify/alias/Zh2fzQ/status with no colon
> 
> not like this http://localhost:5000/minify/alias/:Zh2fzQ/status
> 
> otherwise u will always get true

I definitely know that. 

Are you on Discord?

**Comment by Danko-chan** [comment]
> > you need to sent a get req like this http://localhost:5000/minify/alias/Zh2fzQ/status with no colon
> > not like this http://localhost:5000/minify/alias/:Zh2fzQ/status
> > otherwise u will always get true
> 
> I definitely know that.
> 
> Are you on Discord?

danko-chan#1871

**Comment by BRAVO68WEB** [comment]
image
image

**Comment by Danko-chan** [comment]
rdfhg
dhfdf

**Comment by BRAVO68WEB** [comment]
I guess it was a server side issue. I never said your code was wrong. Fine my bad. I deleted old models and data. Now it works.

**Review by BRAVO68WEB — APPROVED** [review]
Works Fine

**Comment by BRAVO68WEB** [comment]
@allcontributors please add @Danko-chan for code

**Pull request merged**